frontend/src/useStepTime.js

```diff
@@ -11,6 +11,23 @@ function useStepTime() {
     } 
   });
 
+  function updateVisualThreshold(forceData) {
+    let color = null;
+
+    if (forceData <= 19.00 && forceData >= 18.00) {
+      color = "yellow";
+    } else if (forceData < 18.00) {
+      color = "red";
+    } else {
+      color = "green";
+    }
+  
+    const elements = document.querySelectorAll(".CurrentStepTime li");
+    elements.forEach(element => {
+      element.style.background = color;
+    });
+  }
+
```

Comment:
Consider refactoring the updateVisualThreshold function to avoid direct DOM manipulation. Instead of using document.querySelectorAll and modifying the elements' styles, manage the color in the React state. This allows React to handle DOM updates efficiently and ensures your code follows React's declarative approach to rendering.

To do this, instead of changing the color directly in the DOM, you should:

1. Store the color in a React state (for example, colorState).
2. Let React use that state to update the color automatically when the state changes.
By doing this, React takes care of updating the webpage for you in the most efficient way. This also makes your code easier to manage and more in line with React's best practices.

Reply:
Building on the previous suggestion: 
 The current approach directly accesses and updates the DOM elements by using `document.querySelectorAll` and manually changing the color of elements. However, React recommends using its state to manage changes in UI. By handling the color change in a state variable, React can efficiently re-render the components as needed. This makes the code more predictable, easier to debug, and keeps it aligned with React’s declarative nature.

How to Achieve It:
1.Define a State for Color: Create a state variable to hold the color.

`const [boxColor, setBoxColor] = useState("green"); `

2.  Update State Instead of the DOM: In updateVisualThreshold, use setBoxColor to update this state variable based on forceData rather than changing the DOM directly.

`function updateVisualThreshold(forceData) {
    if (forceData <= 19.00 && forceData >= 18.00) {
        setBoxColor("yellow");
    } else if (forceData < 18.00) {
        setBoxColor("red");
    } else {
        setBoxColor("green");
    }
}`

3. Apply the Color in the Component JSX: In your JSX, apply boxColor as the background color of the elements directly. This will allow React to re-render the elements whenever boxColor changes.

`
return (
    <ul className="CurrentStepTime">
        <li style={{ background: boxColor }}>Left Step Time: {stepTime.left}</li>
        <li style={{ background: boxColor }}>Right Step Time: {stepTime.right}</li>
    </ul>
);
`
By following these steps, React will manage the color changes, and you avoid directly interacting with the DOM, making the component more reusable and React friendly. It's not strictly necessary, but it's highly recommended.

Reply:
Thanks for that, I will makes these changes

Reply:
This doesn't seem do work due to how we are rendering it, once I move everything into one place on next weeks pr I will try to implement this